### DISCUSSION TITLE
How to encrypt column ?

### DISCUSSION THREAD

**Discussion by schaulet** [collerek/ormar#109] [Q&A] [answered]
Hi, 

How can we encrypt data of a column ? 

In sqlalchemy there is something like this:
```
username = db.Column(EncryptedType(db.String, _key), nullable=True)
```
Is there a similar mechanism with ormar ?

Thanks.

**Comment by collerek** [comment]
Hi,

No, currently there is no mechanism like this (although you can always encrypt the value before assigning I assume that's not what you want.

Do you want it to work like the mentioned `EncryptedType` from `sqlalchemy_utils` (so decrypt the data on save end decrypt on fetch from db) to basically store encrypted value in db? 

Or something like `pydantic` secret types (https://pydantic-docs.helpmanual.io/usage/types/#secret-types - so stays encrypted/hidden for end users on read and `json()/dict()`)?

**Reply by schaulet** [reply]
The goal is to store crypted data, and manage it transparently inside the app.
So a system like sqlalchemy sounds good.
It for storing sensible data regarding privacy law and legal constraints.

Your second proposition is more like the discussion   #101 about using different models.

**Reply by collerek** [reply]
Sounds like a good idea, can you open enhancement issue for Encrypted field type. 

Since after encryption it's gonna be a hash anyway I assume only one field based on `String` is sufficient (again like with `sqlalchemy_utils`? Or you want to have different types or a wrapper around any field type? If that should work for any field than maybe it should be a common parameter for all other fields (with optional key). 

Of course if you would like to make a PR all contributors are welcomed :)

**Reply by schaulet** [reply]
I think a parameter on fields of every type should be better, but more difficult (if we encrypt a number, what should be it s type regarding database ...).

I will open an enhancement issue for this.

I just clone the repository to try understanding ormar code. I'm not sure to have enough experiences using python (it's my first project using it), but i will do my best.

**Reply by schaulet** [reply]
Do you have some idea where i could start searching in ormar code for applying an encrypt and decrypt methode ?

**Comment by collerek** [comment, marked as answer]
Added in 0.9.8
